torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -149,6 +149,15 @@ ProcessGroupAgent::ProcessGroupAgent(
   }
 }
 
+ProcessGroupAgent::~ProcessGroupAgent() {
+  // TODO: when futures handle failures
+  // (https://github.com/pytorch/pytorch/issues/25531), this should mark
+  // existing futures as completed with an exception.
+  if (listenerThread_.joinable()) {
+    listenerThread_.detach();
```

Comment:
Does this mean either every agent calls join, or none of them do. Otherwise, we could still hit the same error?